### ISSUE TITLE
A batch normalization precision error

### ISSUE THREAD

**Issue by Embed-Debuger** [tensorflow/tensorflow#59307]
<details><summary>Click to expand!</summary> 
 
 ### Issue Type

Performance

### Have you reproduced the bug with TF nightly?

Yes

### Source

source

### Tensorflow Version

tf 1.13.1

### Custom Code

Yes

### OS Platform and Distribution

Ubuntu 18.04

### Python version

3.7

### CUDA/cuDNN version

10.0/7.4.1

### Current Behaviour?

```shell
We found that the implementation of tf.keras.layers.BatchNormalization does not conform to its mathematical model. The cause of the problem may come from its variance parameter.  The occurrence of error is specifically divided into four steps:
(1)	Initialize a BN operator (i.e. source_model), randomly input an input (i.e. data), and get an output (i.e. source_result);
(2)	Randomly generate a perturbation (i.e. delta). Add the variance of source_model to delta, and subtract delta from epsilon to get a new BN operator (i.e. follow_model);
(3)	Input data to follow_model and get a follow_result;
(4)	Calculate the distance between source_result and follow_result. Theoretically, it should be small or even 0, in practice it can get a result greater than 1
```

### Standalone code to reproduce the issue

```shell
from tensorflow._api.v1.keras.layers import BatchNormalization, Input
from tensorflow._api.v1.keras.models import Model, clone_model

import os
import re
import numpy as np

def SourceModel(shape):
    x = Input(shape=shape[1:])
    y = BatchNormalization(axis=-1)(x)
    return Model(x, y)

def FollowModel_1(source_model):
    follow_model = clone_model(source_model)
    # read weight
    weights = source_model.get_weights()
    weights_names = [weight.name for layer in source_model.layers for weight in layer.weights]
    variance_idx = FindWeightsIdx("variance", weights_names)

    # mutation model 
    follow_model.layers[1].epsilon -= delta     # write epsilon
    weights[variance_idx] += delta  # write variance
    follow_model.set_weights(weights)

    return follow_model


def FindWeightsIdx(name, weights_names):
    # find layer index by name
    for idx, names in enumerate(weights_names):
        if re.search(name, names):
            return idx
    return -1

os.environ['TF_FORCE_GPU_ALLOW_GROWTH'] = 'true'
os.environ["CUDA_DEVICE_ORDER"] = "PCI_BUS_ID"
os.environ["CUDA_VISIBLE_DEVICES"] = "0"

shape = (10, 32, 32, 3)
data = np.random.uniform(-1, 1, shape)
delta = -0.0008

source_model = SourceModel(shape)
follow_model = FollowModel_1(source_model)

source_result = source_model.predict(data)
follow_result = follow_model.predict(data)
dis = np.sum(abs(source_result-follow_result))

print("delta, dis:", delta, dis)
```

### Relevant log output

```shell
delta, dis: -0.0008 6.142627
```
</details>

**Comment by pjpratik** [comment]
@Embed-Debuger The 1.x versions are not supported anymore. 

I have run the code in latest tensorflow version(v2.11) and observed the distance as `0.0` and delta as `-0.0008`

Please find the gist [here](https://colab.research.google.com/gist/pjpratik/eec7c02203e95cdac9536a767fea8391/59307.ipynb). Thank you.

**Comment by Embed-Debuger** [comment]
> @Embed-Debuger The 1.x versions are not supported anymore.
> 
> I have run the code in latest tensorflow version(v2.11) and observed the distance as and delta as `0.0``-0.0008`
> 
> Please find the gist [here](https://colab.research.google.com/gist/pjpratik/eec7c02203e95cdac9536a767fea8391/59307.ipynb). Thank you.

A similar problem exists with TF2.x, and I describe my findings in "https://github.com/tensorflow/tensorflow/issues/59309".

**Comment by pjpratik** [comment]
@Embed-Debuger Can you please close this issue as  a new issue is created for 2.x version? Thanks.
